test/driver.js

```diff
@@ -90,6 +113,109 @@ var rasterizeTextLayer = (function rasterizeTextLayerClosure() {
 })();
 
 /**
+ * @class
+ */
+var rasterizeAnnotationLayer = (function rasterizeAnnotationLayerClosure() {
+  var SVG_NS = 'http://www.w3.org/2000/svg';
+
+  var annotationLayerStylePromise = null;
+  function getAnnotationLayerStyle() {
+    if (annotationLayerStylePromise) {
+      return annotationLayerStylePromise;
+    }
+    annotationLayerStylePromise = new Promise(function (resolve) {
+      var xhr = new XMLHttpRequest();
+      xhr.open('GET', './annotation_layer_test.css');
+      xhr.onload = function () {
+        resolve(xhr.responseText);
+      };
+      xhr.send(null);
+    });
+    return annotationLayerStylePromise;
+  }
+
+  function inlineAnnotationImages(images) {
+    var imagePromises = [];
+    for (var i = 0, ii = images.length; i < ii; i++) {
+      var imagePromise = new Promise(function(resolve) {
+        var xhr = new XMLHttpRequest();
+        xhr.responseType = 'blob';
+        xhr.onload = function() {
+          var reader = new FileReader();
+          reader.onloadend = function() {
+            resolve(reader.result);
+          };
+          reader.readAsDataURL(xhr.response);
+        };
+        xhr.onerror = function() {
+          resolve('');
+        };
+        xhr.open('GET', images[i].src);
+        xhr.send();
+      });
+      imagePromises.push(imagePromise);
+    }
+    return imagePromises;
+  }
+
+  function rasterizeAnnotationLayer(ctx, viewport, annotations, page) {
+    return new Promise(function (resolve) {
+      // Building SVG with size of the viewport.
+      var svg = document.createElementNS(SVG_NS, 'svg:svg');
+      svg.setAttribute('width', viewport.width + 'px');
+      svg.setAttribute('height', viewport.height + 'px');
+
+      // Adding element to host our HTML (style + annotation layer div).
+      var foreignObject = document.createElementNS(SVG_NS, 'svg:foreignObject');
+      foreignObject.setAttribute('x', '0');
+      foreignObject.setAttribute('y', '0');
+      foreignObject.setAttribute('width', viewport.width + 'px');
+      foreignObject.setAttribute('height', viewport.height + 'px');
+      var style = document.createElement('style');
+      var stylePromise = getAnnotationLayerStyle();
+      foreignObject.appendChild(style);
+      var div = document.createElement('div');
+      div.className = 'annotationLayer';
+
+      // Rendering annotation layer as HTML.
+      stylePromise.then(function (styles) {
+        style.textContent = styles;
+
+        var annotation_viewport = viewport.clone({ dontFlip: true });
+        PDFJS.AnnotationLayer.render(annotation_viewport, div, annotations,
+                                     page, new LinkServiceMock());
```

Comment:
Just a reminder: With PR #6770 landed, the parameters now need to be passed in as an object instead.